apollo-core/src/main/java/com/spotify/apollo/core/GuiceSpringBridge.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * -\-\-
+ * Spotify Apollo Service Core (aka Leto)
+ * --
+ * Copyright (C) 2013 - 2021 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+package com.spotify.apollo.core;
+
+import org.springframework.boot.SpringBootConfiguration;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.guice.annotation.EnableGuiceModules;
+
+/** Glue to knit together Guice, Spring and make the Spring component scan work. */
+@Configuration
+@EnableGuiceModules
+@SpringBootConfiguration
+@EnableAutoConfiguration
+@ComponentScan(basePackages = "com.spotify")
```

Comment:
I took a stab at fixing this a more generic way in 81cccfb
This makes the Spring bridge look for a `@SpringBootApplication`-annotated class to add as a source when initializing. 
See commit message for some questions, @helenaAbergOstlund @Mandalorian007

Reply:
I like it!

> The main drawback is that the `main` class needs to be annotated with `@SpringBootApplication`, but on the other hand this makes the application a more by-the-book Spring Boot app.
> On the other hand this could be a feature: if we find an app annotated with `@SpringBootApplication` we auto-activate the Spring support, only skipping that if `apollo.spring.enabled` is explicitly set to `false`

Opting for the latter - assuming an app annotated with '@SpringBootApplication' will want to enable spring support - makes a lot of sense. I guess we need to change the code in that case so we always check the stack trace and scan for the annotation, and not only when the flag is enabled.

Reply:
> Opting for the latter - assuming an app annotated with '@SpringBootApplication' will want to enable spring support - makes a lot of sense. I guess we need to change the code in that case so we always check the stack trace and scan for the annotation, and not only when the flag is enabled.

I agree: it makes more sense to try to follow the normal way of activating Spring as far as possible. One question is how to make this work with Fleet Management, though - it would require code changes besides just version bumping to get Spring support in all services. That might not be a real problem, though - I assume it's not too hard to create a fleet change that inspects the main class and adds the annotation if required. But this might be a problem for the future either way - we might cross that bridge when we need it :)